docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,309 @@
+
+Sign the model assertion
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next, we need to sign the model assertion.
+Refer to the guide on `signing model assertion`_ for details on how to sign the model assertion. 
+Below are the needed steps:
+
+1. Create and register a key
+
+   Use ``snapcraft list-keys`` to check your existing keys.
+   If you don't already have a key, create one locally and register it with your account.
+   Choose a meaningful name for the key, as it can be used to sign multiple models.
+   In this guide we use ``rtu-model`` to sign all our real-time Ubuntu models.
+
+   .. code-block:: shell
+
+      snapcraft create-key rtu-model
+      snapcraft register-key rtu-model
+
+   Remember to update the model assertion's ``timestamp``, if you created a new key and plan to use it next.
+
+2. Sign the model assertion
+
+   .. code-block:: shell
+
+      snap sign -k rtu-model model.json > model.signed.yaml
+
+   The ``snap sign`` command takes JSON as input and produces YAML as output.
+
+   .. tip::
+
+      You need to repeat the signing every time you change the input model, because the signature is calculated based on the model.
+
+
+Build the image
+~~~~~~~~~~~~~~~
+
+First, get familiar with the tooling by referring to the guide on `building Ubuntu Core images`_.
+
+We use ``ubuntu-image`` and need to set the paths to the following as input:
+
+- Exported store credentials
+- Signed model assertion YAML file
+
+Export the store credentials to a file:
+
+.. code-block:: shell
+
+    snapcraft export-login credentials.txt
+
+Then build the image:
+
+.. code-block:: console
+
+    $ UBUNTU_STORE_AUTH_DATA_FILENAME=credentials.txt ubuntu-image snap model.signed.yaml --verbose --validation=enforce
+    [0] prepare_image
+    Fetching snapd (21759)
+    Fetching realtime-kernel (149)
+    Fetching core22 (1586)
+    Fetching pc (146)
+    [1] load_gadget_yaml
+    [2] set_artifact_names
+    [3] populate_rootfs_contents
+    [4] generate_disk_info
+    [5] calculate_rootfs_size
+    [6] populate_bootfs_contents
+    [7] populate_prepare_partitions
+    [8] make_disk
+    [9] generate_snap_manifest
+    Build successful
+
+This downloads all the snaps specified in the model assertion and builds an image file called ``pc.img``.
+
+.. hint::
+
+    To fetch the ``realtime-kernel`` snap for this image build, it should be included explicitly in your dedicated Snap Store.
+
+.. code-block:: console
+
+    $ file pc.img 
+    pc.img: DOS/MBR boot sector; partition 1 : ID=0xee, start-CHS (0x0,0,0), end-CHS (0x0,0,0), startsector 1, 6195199 sectors, extended partition table (last)
+
+✅ The image file is now ready. Refer to Ubuntu Core guide on `flashing the image to a storage medium`_.
+
+----
+
+After installing this image on your device, you can continue by tuning your system for real-time processing. 
+The :ref:`ubuntu-core` guide describes the method for dynamically configuring the kernel command line parameters.
```

Comment:
hi @farshidtz , this target doesn't exist.
I think this was where you wanted to link to?
:ref:`ubuntu-core-kernel-cmdline`